pinot-core/src/main/java/com/linkedin/pinot/core/operator/aggregation/groupby/DefaultGroupByExecutor.java

```diff
@@ -123,11 +130,22 @@ public void process(int[] docIdSet, int startIndex, int length) {
     Preconditions
         .checkState(_inited, "Method 'process' cannot be called before 'init' for class " + getClass().getName());
 
-    _singleValueBlockCache.initNewBlock(docIdSet, startIndex, length);
+    // We have already reached max number of keys, punt on further processing.
+    if (_groupKeyGenerator.getCurrentGroupKeyUpperBound() > _numGroupsLimit) {
```

Comment:
I don't see it as adding much value, as it will potentially keep the threads busy, and the results will be inaccurate anyways. How about just throwing an exception instead of partial results?

Also, I explored the option of doing so, and it seems to pollute at least three class that will now have to handle the case when we have hit this limit, and start seeing new keys that were not already seen.
We need a clean solution to handle such cases, but that would be a bigger change. As we discussed offline below are a couple of proposals to explore, both of which are considerable changes:
* Go off heap (mmap) to compute the group-by's.
* Use two pass techniques:
     * In first pass we map a large range of group keys into a much smaller range (multiple keys may map on to same value), and compute aggregation on these compressed key ranges. Sort the keys based on aggregations function values, and trim the ones with smaller values.
      * In the second pass, do the actual group-by, while skipping keys that were trimmed in the first pass.

Given these are non-trivial changes, we need to put a temporary stop on the fire until we have  a clean solution. This PR is just for the temporary stop.